Added all-your-base exercise to JavaScript track

javaeeeee opened the pull request:
Added all-your-base exercise to the JavaScript track

rchavarria commented on exercises/all-your-base/all-your-base.spec.js:
```diff
@@ -0,0 +1,140 @@
+
+  it('wrong input_base base less than two', function () {
+    expect(function () {
+      converter.convert([0]);
```
To reflect better what line 9 describes, shouldn't this line be:

    converter.convert([0], 1);

Reading the test, it looks like `[0]` is the `inputBase` parameter, but in fact, it's `array` parameter.

What do you think? Do you think users will understand better this test adding the `inputBase` parameter as `1` (or another value less than `2`)?

rchavarria commented on exercises/all-your-base/all-your-base.spec.js:
```diff
@@ -0,0 +1,140 @@
+
+  xit('wrong output_base base less than two', function () {
+    expect(function () {
+      converter.convert([0], 2);
```
Same as comment on line 11

rchavarria commented:
Awesome!

It's nice to have a new exercise in this track.

After talking about comments on lines 11 and 23, would you mind to squash your commits? Would you need some help for doing that?

javaeeeee commented:
Hello Ruben,

Thank you for reviewing the code. I would like to mention that the tests
you commented on were intended to test the cases of missing bases as
JavaScript allows to pass as little arguments as you like. :)

There are separate tests that test unity as a base like 'first base is one'
and 'second base is one' on lines 99 and 105.

If you thank that testing the case of missing arguments is redundant, I can
remove them altogether. Or maybe, it's better to change the description of
the tests?

Thank you.

On Thu, Dec 1, 2016 at 5:24 AM, Rubén Chavarría <notifications@github.com>
wrote:

> Awesome!
>
> It's nice to have a new exercise in this track.
>
> After talking about comments on lines 11 and 23, would you mind to squash
> your commits? Would you need some help for doing that?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/exercism/xjavascript/pull/308#issuecomment-264135397>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIezhwUbaWMCVkQ7dyr85FhJrq_HeC8zks5rDqBwgaJpZM4LAtci>
> .
>

rchavarria commented:
You can keep tests without the parameter that are currently on L9 and L21. They show a specific feature of JavaScript. But, I would move the close to the ones you mentioned on L99 and L105 (I didn't notice them). And maybe, a better description could be `missing input/output base throws an error` or similar.

javaeeeee commented:
Hello Ruben,

Thank you for your comments. Renamed and moved the tests checking missing
inputs. Also squashed the commits.

On Sat, Dec 3, 2016 at 12:35 PM, Rubén Chavarría <notifications@github.com>
wrote:

> You can keep tests without the parameter that are currently on L9 and L21.
> They show a specific feature of JavaScript. But, I would move the close to
> the ones you mentioned on L99 and L105 (I didn't notice them). And maybe, a
> better description could be missing input/output base throws an error or
> similar.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/exercism/xjavascript/pull/308#issuecomment-264653341>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIezh3sgjNlRE9LDHvEFiScJsZRBMwVhks5rEahhgaJpZM4LAtci>
> .
>

rchavarria commented:
Thanks so much for this new exercise.

rchavarria commented:
I'm going to close this PR manually. I'm not sure why it wasn't marked as *merged* automatically when I pushed the merge :cry:

javaeeeee commented:
Thank you.

On Wed, Dec 7, 2016 at 12:37 PM, Rubén Chavarría <notifications@github.com>
wrote:

> I'm going to close this PR manually. I'm not sure why it wasn't marked as
> *merged* automatically when I pushed the merge 😢
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/exercism/xjavascript/pull/308#issuecomment-265515865>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIezh5ySDdaePJ3Eys3EWqJXkx2i4I6gks5rFu7egaJpZM4LAtci>
> .
>
